MarsBaseBuilder/Renderer.fs

```diff
@@ -0,0 +1,37 @@
+﻿module MarsBaseBuilder.Renderer
+
+open Microsoft.Xna.Framework
+
+open MarsBaseBuilder.Measures
+
+type DrawCommand = 
+    | Background
+    | Rectangle of a : ScreenPoint * b : ScreenPoint
+
+let internal mapToScreenPoint (p : Position) : ScreenPoint =
+    { x = 400<screenPx> + (int p.x) * 1<screenPx>
+      y = 300<screenPx> + (int p.y) * 1<screenPx> }
+
+let internal offset x y (sp : ScreenPoint) =
+    { sp with x = sp.x + x; y = sp.y + y }
+
+let baseRadius = 5<screenPx>
+let internal draw coords = function
+    | Base -> 
+        let pos = mapToScreenPoint coords
+        Rectangle (offset (-baseRadius) (-baseRadius) pos, offset baseRadius baseRadius pos)
+
+let commands (state : GameLogic.GameState) : ResizeArray<DrawCommand> =
+    let list = ResizeArray()
```

Comment:
Yes, it is. Everything that get called from `Draw` method is extremely performance-sensitive and should generate as little garbage as possible. I don't intend to actually modify this `ResizeArray` outside of this function, but we need it to keep its type due to performance considerations (`System.List`'s `GetEnumerator` returns `struct` and that's important).

You may argue that it's a premature optimization, but I think that current setup should be okay for a game engine. I have a fear that we'll need to optimize it more in the future.